src/string_search.h

```diff
@@ -0,0 +1,597 @@
+// Copyright 2011 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SRC_STRING_SEARCH_H_
+#define SRC_STRING_SEARCH_H_
+
+#include "node.h"
+#include <string.h>
+
+namespace node {
+namespace stringsearch {
+
+// Returns the maximum of the two parameters.
+template <typename T> T Max(T a, T b) { return a < b ? b : a; }
+
+static const uint32_t kMaxOneByteCharCodeU = 0xff;
+
+static inline int NonOneByteStart(const uint16_t *chars, int length) {
+  const uint16_t *limit = chars + length;
+  const uint16_t *start = chars;
```

Comment:
It's not a bad idea but it only works well if all developers use the same version of clang-format,  there are quite a few differences between 3.5 and 3.6.

Apropos the style, I think the Google style is closer to what we use than Chromium's is.